test/interface/static_array_tests.jl

```diff
@@ -17,7 +17,7 @@ sol = solve(ode, Tsit5())
 sol = solve(ode, Vern9())
 @test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
 
-u0 = [fill(2, SVector{2, Float64}), ones(SVector{2, Float64})]
+u0 = VectorOfArray([fill(2, SVector{2, Float64}), ones(SVector{2, Float64})])
```

Comment:
Does this mean that plain `Array`s of `SVector`s will not be supported anymore and we have to switch to `VectorOfArray`?

Reply:
See https://github.com/SciML/SciMLBase.jl/pull/561. Basically I am trying to get things to a well-defined interface. This little bit sticks out as sore area of DiffEq where we don't throw any errors about arrays of arrays since 10 out of like 300 methods supports arrays of arrays. But we should be able to support such types of targets more broadly without overhead by just using a small wrapper that makes everything conform to the interface. If this is allowed, then we can throw informative errors as to what's going on and we can vastly clean up a lot of the code to not require hacks for specific methods to avoid assumptions about the lack of Number eltype. Since it's an undocumented bit only used downstream by you and @Datseris, it's a small discussion to be had with many good consequences to most users, and would make things like transitioning a code to implicit methods more readily available since the Vector of Vector form will likely never be able to support linear algebra appropriately. 

As you can see from the PR, there is a keyword argument being made to bypass the interface erroring, but I hope that is only used during development and is not a crutch that is actually relied on for these kinds of tasks given that there should be zero overhead to using a more structured and interface-compliant form.

Reply:
We use arrays of `SVector`s at least for some solvers in Trixi.jl with explicit Runge-Kutta methods. If OrdinaryDiffEq.jl transitions to another interface requirement, it would be great to get a breaking release describing this change. 

We would also need more interfaces such as `resize!` and `pointer` for `VectorOfArray`.

Reply:
> As you can see from the PR, there is a keyword argument being made to bypass the interface erroring, but I hope that is only used during development and is not a crutch that is actually relied on for these kinds of tasks given that there should be zero overhead to using a more structured and interface-compliant form.

Which PR do you refer to? I could only see changes to the tests here and new docs at https://github.com/SciML/SciMLBase.jl/pull/561

Reply:
I'm not sure undocumented side pieces really constitute breaking. We are trying to impose such interfaces so that there is a strong sense of what is actually breaking. 

> We would also need more interfaces such as resize! and pointer for VectorOfArray.

Add suggestions in the interface PR and I'll happily accept that.

Reply:
That PR describes the keyword argument.

Reply:
I see, thanks.

Reply:
I only use vector of static vectors internally at a rather specific type of dynamical system. If the change required is to simply change the input to the wrapper `VectorOfArray` that doesn't bother me, I would really only have to change 2 lines of code in the whole library. It is also not exposed to the user. Simply let me know when, and at which release, I should make the change to `VectorOfArray` and that's not a problem!

Reply:
Does this apply to any `AbstractVector{<:SVector}` type (e.g., StructArrays) as well?

Reply:
In BVDiffEq we were using a `AbstractArray{<:AbstractArray}` for specifying the initial guess on a mess. We can upperbound SciMLBase there and cut a breaking release.

Reply:
Is bounding SciMLBase compat enough to address the impact of this change?

Reply:
Not SciMLBase my bad. DiffEqBase needs a upperbound. See https://github.com/SciML/BoundaryValueDiffEq.jl/pull/152

Reply:
> In BVDiffEq we were using a AbstractArray{<:AbstractArray} for specifying the initial guess on a mess. We can upperbound SciMLBase there and cut a breaking release.

BVDiffEq we need to specialize there because that's a legitimate use case, that's quite different.

Reply:
How do you plan to do that? We may use it as inspiration for us in Trixi.jl.